### PR TITLE
Do common node config on bwc tests

### PR DESCRIPTION
**Pull request by alpar-t**
Before this PR we always ever ran `ElasticsearchCluster.start` once, and
the common node config was never done.
This becomes apparent in upgrading from `6.x` to `7.x` as the new config
is missing preventing the cluster from starting.

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java [hunk 1]
```diff
@@ -259,6 +264,10 @@ public void start() {
             // Can only configure master nodes if we have node names defined
             if (nodeNames != null) {
                 if (node.getVersion().onOrAfter("7.0.0")) {
+                    node.defaultConfig.keySet().stream()
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java]: I'm pretty sure that, with the exception of `initial_master_nodes` all this can, and probably should, go in `ElasticsearchNode#start()`.

**Review by mark-vieira — CHANGES_REQUESTED** [review]
When we can we should prefer putting this kind of logic in `ElasticsearchNode` vs `ElasticsearchCluster`.

In any case, we need to address refactoring all this stuff to try and separate configuration from node-handling logic ASAP. All this handling of version-specific edge cases is starting to compound and make things hard to follow.

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java (hunk 1 above)
Comment (atorok) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java]: Technically yes, but these are settings relating to how the cluster forms that relate to each-other,
so the settings are per node, but relate to the cluster itself. 
Granted we could pull other such settings up to the cluster level. 
I agree on your points on refactoring but would like to complete the back-port first.

**Review by mark-vieira — APPROVED** [review]
